src/app/modules/zxing-scanner/browser-code-reader.ts

```diff
@@ -197,12 +209,20 @@ export class BrowserCodeReader {
    * @param videoElement
    * @param callbackFn
    */
-  private bindEvents(videoElement: HTMLVideoElement, callbackFn?: (result: Result) => any): void {
-
-    if (typeof callbackFn !== 'undefined') {
-      this.videoPlayingEventListener = () => this.decodingStream = this.decodeWithDelay(this.timeBetweenScans)
-        .pipe(catchError((e, x) => this.handleDecodeStreamError(e, x)))
-        .subscribe((x: Result) => callbackFn(x));
+  private bindEvents(videoElement: HTMLVideoElement): void {
+
+    if (this.onDecodeComplete) {
+      this.videoPlayingEventListener = () => this.decodeWithDelay(this.timeBetweenScans)
+        .pipe(
+          catchError((e, x) => this.handleDecodeStreamError(e, x)),
+          takeUntil(this.unsubscribe$),
+        )
+        .subscribe((x: Result) => {
```

Comment:
I would like to expose this stream or some `BehaviorSubject` updated by it, so the user could directly subscribe to it without passing callback functions to the component.